### ISSUE TITLE
Cannot publish ethpm package with dependencies

### ISSUE THREAD

**Issue by alex-miller-0** [trufflesuite/truffle#699]
## Issue

Publishing an ethpm package (`truffle publish`) works nicely with self-contained contracts (i.e. those that do not import any other contracts), but I can't get it to work with contract imports.

## Steps to Reproduce

I am importing a `Token` contract [here](https://github.com/alex-miller-0/eth-relayer/blob/master/contracts/TrustedRelay.sol#L12):

```
import "./Token.sol"; 
```

*Note: importing from an ethpm install (i.e. `import "tokens/Token.sol"`) has the same effect.*

I instantiate a token in a function call [here](https://github.com/alex-miller-0/eth-relayer/blob/master/contracts/TrustedRelay.sol#L64):

```
Token t;
t = Token(tokenAddress);
```

`truffle compile`, `truffle migrate`, and `truffle deploy` all work fine.

## Expected Behavior

`truffle publish` should publish this package to ethpm

## Actual Results

I get the following error:

```
ParsedContract.sol:64:5: DeclarationError: Identifier not found or not unique.
    Token t;
    ^---^
```

## Environment

* Operating System: OSX
* Truffle version: 4.0.1
* Ethereum client: INFURA (ropsten)
* node version: 8.6.0
* npm version: 5.3.0

**Comment by Vourhey** [comment]
I'm having the same error! Still no response? 
as OP said compile works fine, but it's impossible to publish

**Comment by frangio** [comment]
We're getting the same error in the OpenZeppelin repository. It started happening around version 1.4.0.

Does anyone from the Truffle team have any clue as to what could be happening?

**Comment by cgewecke** [comment]
@frangio Are you seeing this when trying publish Zeppelin itself, or when trying to publish something that imports contracts from an npm installed zeppelin? Or from ethpm installed Zeppelin.

There is a [recent fix](https://github.com/trufflesuite/truffle-compile/pull/48) for this that's pending release (early next week) but it really only addresses the case of contracts that rely on ethpm dependencies.  

If you have reproduction steps for the Zeppelin case, will investigate.

**Comment by frangio** [comment]
When trying to publish OpenZeppelin itself. Can you try running `truffle publish` on the [OpenZeppelin/zeppelin-solidity](https://github.com/OpenZeppelin/zeppelin-solidity) repo on your machine and see if that fails?

**Comment by cgewecke** [comment]
@frangio Yes I can see it failing and see why in the code as well. We'll try to fix this ASAP, thanks for pinging this.

**Comment by frangio** [comment]
@cgewecke Thanks to you for the quick response as always! ♥️

**Comment by cgewecke** [comment]
@alex-miller-0 @frangio This should be fixed in `4.1.5`. Please report back if you continue to see this or have other issues publishing to ethpm. 

Thanks!!
